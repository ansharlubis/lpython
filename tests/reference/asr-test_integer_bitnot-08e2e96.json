{
    "basename": "asr-test_integer_bitnot-08e2e96",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/test_integer_bitnot.py",
    "infile_hash": "dc976a358fbeebedead889f8e85b3eed4ac77ee20f68fcac58b81429",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-test_integer_bitnot-08e2e96.stdout",
    "stdout_hash": "494e241d7d85529a8f8aacd4715a24f510ccd2adda07cb6f61afc57e",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}