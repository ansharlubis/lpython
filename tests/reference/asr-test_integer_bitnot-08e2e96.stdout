([1m[35mTranslationUnit[39m[0m 
   ([33mSymbolTable[39m 
      1 
      {
         f: 
            ([1m[35mSubroutine[39m[0m 
               ([33mSymbolTable[39m 
                  2 
                  {
                     i: 
                        ([1m[35mVariable[39m[0m 
                           2 
                           i [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mInteger[39m[0m 4 []) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.), 
                     res: 
                        ([1m[35mVariable[39m[0m 
                           2 
                           res [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mInteger[39m[0m 4 []) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.)
                  }) 
               f [] [
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33mi[39m) ([1m[35mIntegerConstant[39m[0m [36m5[39m 
                  ([1m[35mInteger[39m[0m 4 [])) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33mres[39m) 
                  ([1m[35mIntegerBitNot[39m[0m 
                     ([1m[35mVar[39m[0m 2 [33mi[39m) 
                     ([1m[35mInteger[39m[0m 4 []) ()) ()) 
               ([1m[35mAssert[39m[0m 
                  ([1m[35mIntegerCompare[39m[0m 
                     ([1m[35mVar[39m[0m 2 [33mres[39m) [1m[32m
                     Eq[39m[0m 
                     ([1m[35mIntegerUnaryMinus[39m[0m ([1m[35mIntegerConstant[39m[0m [36m6[39m 
                        ([1m[35mInteger[39m[0m 4 [])) 
                        ([1m[35mInteger[39m[0m 4 []) ([1m[35mIntegerConstant[39m[0m [36m-6[39m 
                        ([1m[35mInteger[39m[0m 4 []))) 
                     ([1m[35mLogical[39m[0m 4 []) ()) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33mi[39m) 
                  ([1m[35mIntegerUnaryMinus[39m[0m ([1m[35mIntegerConstant[39m[0m [36m235346[39m 
                     ([1m[35mInteger[39m[0m 4 [])) 
                     ([1m[35mInteger[39m[0m 4 []) ([1m[35mIntegerConstant[39m[0m [36m-235346[39m 
                     ([1m[35mInteger[39m[0m 4 []))) ()) 
               ([1m[35mAssert[39m[0m 
                  ([1m[35mIntegerCompare[39m[0m 
                     ([1m[35mIntegerBitNot[39m[0m 
                        ([1m[35mVar[39m[0m 2 [33mi[39m) 
                        ([1m[35mInteger[39m[0m 4 []) ()) [1m[32m
                     Eq[39m[0m ([1m[35mIntegerConstant[39m[0m [36m235345[39m 
                     ([1m[35mInteger[39m[0m 4 [])) 
                     ([1m[35mLogical[39m[0m 4 []) ()) ())] [1m[32m
               Source[39m[0m [1m[32m
               Public[39m[0m [1m[32m
               Implementation[39m[0m () .false. .false.)
      }) 
   [
   ([1m[35mSubroutineCall[39m[0m 1 [33mf[39m () [] ())])
(TranslationUnit (SymbolTable 1 {_lpython_main_program: (Subroutine (SymbolTable 4 {}) _lpython_main_program [] [(SubroutineCall 1 f () [] ())] Source Public Implementation () .false. .false.), f: (Subroutine (SymbolTable 2 {i: (Variable 2 i Local () () Default (Integer 4 []) Source Public Required .false.), res: (Variable 2 res Local () () Default (Integer 4 []) Source Public Required .false.)}) f [] [(= (Var 2 i) (IntegerConstant 5 (Integer 4 [])) ()) (= (Var 2 res) (IntegerBitNot (Var 2 i) (Integer 4 []) ()) ()) (Assert (IntegerCompare (Var 2 res) Eq (IntegerUnaryMinus (IntegerConstant 6 (Integer 4 [])) (Integer 4 []) (IntegerConstant -6 (Integer 4 []))) (Logical 4 []) ()) ()) (= (Var 2 i) (IntegerUnaryMinus (IntegerConstant 235346 (Integer 4 [])) (Integer 4 []) (IntegerConstant -235346 (Integer 4 []))) ()) (Assert (IntegerCompare (IntegerBitNot (Var 2 i) (Integer 4 []) ()) Eq (IntegerConstant 235345 (Integer 4 [])) (Logical 4 []) ()) ())] Source Public Implementation () .false. .false.), main_program: (Program (SymbolTable 3 {}) main_program [] [(SubroutineCall 1 _lpython_main_program () [] ())])}) [])
